Add support for 'elif'

corbinbs opened the pull request:
While using https://github.com/ericclemmons/jinja.js, I ran into errors with some of my macros that contained code similar to:

{% if condition1 %}
A
{% elif condition2 %}
B
.
.
.
{% endif %}

Please consider adding these changes to support that jinja syntax.

corbinbs committed:
Add support for 'elif' as seen here: http://jinja.pocoo.org/docs/templates/#if

ceymard commented:
Quick question : Why not just rename the node elseif to elif since it would seem they do pretty much the same thing ?

Or just adding elif: NodeElseIf in the exports ?

It seems there is a lot of code duplication otherwise. Your thoughts ?

corbinbs commented:
I was originally going with the rename approach (replacing the elseif with elif) but then I noticed the jinja.js project listed support for Twig and http://twig.sensiolabs.org/doc/tags/if.html shows elseif being used.  
I'm really only concerned with jinja but I didn't want to break something others might be depending on.
So, I dropped back and just added a new node to (hopefully) avoid breaking anything.  
I like renaming elseif to elif if elseif is not really needed.

ceymard commented:
Ok. I like your patch.

Thanks.
